api/src/pcapi/core/providers/titelive_book_search.py

```diff
@@ -0,0 +1,240 @@
+    gtl_id = most_precise_genre
+    return gtl_id.code
+
+
+def get_ineligibility_reason(article: TiteLiveBookArticle, title: str) -> str | None:
+    # Ouvrage avec pierres ou encens, jeux de société ou escape game en coffrets,
+    # marchandisage : jouets, goodies, peluches, posters, papeterie, etc...
+    article_tva = float(article.taux_tva) if article.taux_tva else None
+    if article_tva == float(constants.BASE_VAT):
+        return "vat-20"
+
+    if article_tva == float(constants.PAPER_PRESS_VAT) and article.codesupport == constants.PAPER_PRESS_SUPPORT_CODE:
+        return "press"
+
+    if article.codesupport == constants.CALENDAR_SUPPORT_CODE:
+        return "calendar"
+
+    if article.codesupport == constants.POSTER_SUPPORT_CODE:
+        return "poster"
+
+    if article.codesupport == constants.PAPER_CONSUMABLE_SUPPORT_CODE:
+        return "paper-consumable"
+
+    # Coffrets (contenant un produit + un petit livret)
+    if article.codesupport == constants.BOX_SUPPORT_CODE:
+        return "box"
+
+    # Oracles contenant des jeux de tarot
+    if article.codesupport == constants.OBJECT_SUPPORT_CODE:
+        return "object"
+
+    # ouvrage "lectorat 18+" (Pornographie / ultra-violence)
+    if article.id_lectorat == constants.LECTORAT_EIGHTEEN_ID:
+        return "pornography-or-violence"
+
+    # Toeic or toefl
+    if constants.TOEIC_TEXT in title or constants.TOEFL_TEXT in title:
+        return "toeic-toefl"
+
+    # --- GTL-based categorization ---
+    gtl_id = get_gtl_id(article)
+    if gtl_id == EMPTY_GTL.code:
+        return "empty-gtl"
+
+    gtl_level_01_code = gtl_id[:2]
+
+    # ouvrage du rayon scolaire
+    if gtl_level_01_code == constants.GTL_LEVEL_01_SCHOOL:
+        return "school"
+
+    # ouvrage du rayon parascolaire,
+    # code de la route (méthode d'apprentissage + codes basiques), code nautique, code aviation, etc...
+    if gtl_level_01_code == constants.GTL_LEVEL_01_EXTRACURRICULAR:
+        return "extracurricular"
+
+    # Petite jeunesse (livres pour le bains, peluches, puzzles, etc...)
+    gtl_level_02_code = gtl_id[2:4]
+    if gtl_level_01_code == constants.GTL_LEVEL_01_YOUNG and gtl_level_02_code in [
+        constants.GTL_LEVEL_02_BEFORE_3,
+        constants.GTL_LEVEL_02_AFTER_3_AND_BEFORE_6,
+    ]:
+        return "small-young"
+
+    # --- Default categorization ---
+    if article.codesupport not in constants.TITELIVE_BOOK_SUPPORTS_BY_CODE:
+        # If we find no known support code, the product is deemed ineligible
+        return "no-known-codesupport"
+
+    if not constants.TITELIVE_BOOK_SUPPORTS_BY_CODE[article.codesupport].get("is_allowed"):
+        return "uneligible-product-subcategory"
+
+    return None
+
+
+def get_book_format(codesupport: str | None) -> str | None:
+    if codesupport is None:
+        return None
+    book_format = constants.TITELIVE_BOOK_SUPPORTS_BY_CODE[codesupport].get("book_format")
```

Comment:
[détail] Idem plus haut : y a-t-il une raison pour ne pas avoir un champ `book_format` ?

```suggestion
    book_format = constants.TITELIVE_BOOK_SUPPORTS_BY_CODE[codesupport]["book_format"]
```

commit : "feat: use titelive API for books"